src/main/java/io/axoniq/axonserver/connector/event/impl/DcbEventChannelImpl.java

```diff
@@ -0,0 +1,225 @@
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * {@link DcbEventChannel} implementation, serving as the event connection between Axon Server and a client
+ * application.
+ *
+ * @author Milan Savic
+ * @since 2024.1.0
+ */
+public class DcbEventChannelImpl extends AbstractAxonServerChannel<Void> implements DcbEventChannel {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DcbEventChannelImpl.class);
+
+    private static final int BUFFER_SIZE = 32;
+    private static final int REFILL_BATCH = 8;
+    private final DcbEventStoreGrpc.DcbEventStoreStub eventStore;
+    private final ClientIdentification clientIdentification;
+
+    /**
+     * Instantiate an {@link AbstractAxonServerChannel}.
+     *
+     * @param clientIdentification     the identification of the client
+     * @param executor                 a {@link ScheduledExecutorService} used to schedule reconnections
+     * @param axonServerManagedChannel the {@link AxonServerManagedChannel} used to connect to AxonServer
+     */
+    public DcbEventChannelImpl(ClientIdentification clientIdentification,
+                               ScheduledExecutorService executor,
+                               AxonServerManagedChannel axonServerManagedChannel) {
+        super(clientIdentification, executor, axonServerManagedChannel);
+        this.eventStore = DcbEventStoreGrpc.newStub(axonServerManagedChannel);
+        this.clientIdentification = clientIdentification;
+    }
+
+    @Override
+    public void connect() {
+        // nothing to do here
+    }
+
+    @Override
+    public void reconnect() {
+        // nothing to do here
+    }
+
+    @Override
+    public void disconnect() {
+        // nothing to do here
+    }
+
+    @Override
+    public boolean isReady() {
+        return true;
+    }
+
+    @Override
+    public AppendEventsTransaction startTransaction() {
+        FutureStreamObserver<AppendEventsResponse> response = new FutureStreamObserver<>(null);
+        StreamObserver<AppendEventsRequest> clientStream = eventStore.append(response);
+        return new AppendEventsTransactionImpl(clientStream, response);
+    }
+
+    @Override
+    public ResultStream<StreamEventsResponse> stream(StreamEventsRequest request) {
+        AbstractBufferedStream<StreamEventsResponse, Empty> result =
+                new AbstractBufferedStream<StreamEventsResponse, Empty>(clientIdentification.getClientId(),
+                                                                        BUFFER_SIZE,
+                                                                        REFILL_BATCH) {
+                    @Override
+                    protected Empty buildFlowControlMessage(FlowControl flowControl) {
+                        return null;
```

Comment:
Curious: There's no flow control message for streaming events from a DCB channel (yet)? Is flow control not supported?

Reply:
Flow control is done on the protocol (gRPC) level, rather than on the application level.

Reply:
Cool, seems fair! How would the connector influence the flow control? Not sure if that's going to be a request, just trying to understand the window of opportunities from AF using the Connector.

Reply:
Awesome, then I am completely fine with this structure. I'll resolve the comment right away.